### PR TITLE
[Checkbox] Fix shift click

### LINKED ISSUE #11809: [IndexTable] Shift click (multiselect) is broken
Our checkboxes are custom where the `label` does the actual work when click happens. Firefox doesn't allow the click event to bubble up on the label when shift is held down

https://bugzilla.mozilla.org/show_bug.cgi?id=559506#:~:text=18%20days%20ago-,If%20you%20shift%2Bclick%20or%20ctrl%2Bclick%20on%20a%20label,checkbox%2C%20checkbox%20is%20not%20checked.

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/11809

https://bugzilla.mozilla.org/show_bug.cgi?id=559506#:~:text=18%20days%20ago-,If%20you%20shift%2Bclick%20or%20ctrl%2Bclick%20on%20a%20label,checkbox%2C%20checkbox%20is%20not%20checked.

### WHAT is this pull request doing?

This PR changes our checkbox input from being 1px / 1px hidden, to full size with opactiy 0. So the click event gets triggered.

### How to 🎩

[Spin](https://admin.web.checkbox.kyle-durand.us.spin.dev/store/shop1/orders)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by laurkim — APPROVED** [review]
Nice! 💯 🙌 
Spin and Storybook look good in Chrome/Safari/Firefox

File: .changeset/funny-timers-explode.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+[IndexTable] Fixed shift click bug in Firefox
```
Comment (chloerice) [inline .changeset/funny-timers-explode.md]:
```suggestion
Fixed Shift + click selection of `IndexTable.Row` not working in Firefox
```

**Review by chloerice — APPROVED** [review]
Thanks for fixing this @kyledurand! Tested in Firefox and works well.

**Review by alex-page — APPROVED** [review]

**Pull request merged**